### PR TITLE
refactor(sync): replace usage of remoteID with (given)URL

### PR DESCRIPTION
**Pull request by dskuza**
## Summary

Replaces usage of `remoteID` with `(given)URL` for applicable GraphQL queries and mutations.

## References 

IN-1601

## Implementation Details

Some GraphQL queries and mutations can be replaced to use a URL over a remote ID. This will also improve client-side UX by ensuring that (saved) items that have not yet been sync'd with the server (such as when offline) can still have certain interactions still occur, such as archiving.

The below queries / mutations had their respective updated API with URL usage:

- `updateSavedItemFavorite` -> `savedItemFavorite`
- `updateSavedItemUnFavorite` -> `savedItemUnFavorite`
- `updateSavedItemArchive` -> `savedItemArchive`
- `deleteSavedItem` -> `savedItemDelete`
- `replaceSavedItemTags` -> `savedItemTag`
- `itemByItemId` -> `itemByUrl`

## Test Steps

- [ ] General regression test for item actions: (un)favorite, (un)archive, delete, update tags, opening a syndicated article

## PR Checklist:

- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/Sync/PocketSaveService.swift [hunk 1]
```diff
@@ -159,16 +159,20 @@ public class PocketSaveService: SaveService {
                 }
                 queue.addOperation(operation)
             } else {
-                let mutation = ReplaceSavedItemTagsMutation(input: [SavedItemTagsInput(savedItemId: remoteID, tags: names)])
+                let url = savedItem.item?.givenURL ?? savedItem.url
+                let mutation = SavedItemTagMutation(
+                    input: SavedItemTagInput(givenUrl: url, tagNames: names),
+                    timestamp: ISO8601DateFormatter.pocketGraphFormatter.string(from: .now)
```
Comment (cyndichin) [inline PocketKit/Sources/Sync/PocketSaveService.swift:165]: curious to hear about the context in behind adding this timestamp to `tag` and why it's needed as an input

**Review by cyndichin — APPROVED** [review]
looks good, I had one curious comment. Overall, don't see any issues with performing the actions under Test Steps. However, I was unable to find a syndicated article with Unified Home. Otherwise, great work!

**Review by Gio2018 — APPROVED** [review]
Looks good to me @dskuza !! Did a quick smoke test and I did not find any issue.

File: PocketKit/Sources/Sync/PocketSaveService.swift (hunk 1 above)
Comment (dskuza) [reply PocketKit/Sources/Sync/PocketSaveService.swift:165]: @cyndichin is it fair to say "because the backend says so"? 😅

**Pull request merged**